app/models/league.rb

```diff
@@ -18,53 +17,58 @@ def to_param
   end
 
   def update_badges
-    self.users.update_all(most_wins: false, top_crawler: false, worst_crawler: false, longest_winning_streak: false, most_teams: false, longest_winning_streak_ever: false )
-
-    most_wins.update_attribute(:most_wins, true) if most_wins && most_wins.number_of_wins > 0
-    top_crawler.update_attribute(:top_crawler, true) if top_crawler && top_crawler.number_of_crawls > 0
-    worst_crawler.update_attribute(:worst_crawler, true) if worst_crawler && worst_crawler.number_of_crawlings > 0
-    longest_winning_streak.update_attribute(:longest_winning_streak, true) if longest_winning_streak && longest_winning_streak.winning_streak > 0
-    most_teams.update_attribute(:most_teams, true) if most_teams
-    longest_winning_streak_ever.update_attribute(:longest_winning_streak_ever, true) if longest_winning_streak_ever
-
+    reset_all_user_badges
+    ensure_user(most_wins).update_attribute(:most_wins, true)
+    ensure_user(top_crawler).update_attribute(:top_crawler, true)
+    ensure_user(worst_crawler).update_attribute(:worst_crawler, true)
+    ensure_user(longest_winning_streak).update_attribute(:longest_winning_streak, true)
+    ensure_user(most_teams).update_attribute(:most_teams, true)
+    ensure_user(longest_winning_streak_ever).update_attribute(:longest_winning_streak_ever, true)
   end
 
-  def top_crawler
-    self.users.order('number_of_crawls desc').order('updated_at desc').take
+  def most_wins
+    users.where('number_of_wins > 0').order('number_of_wins desc, updated_at desc').take
   end
 
-  def most_wins
-    self.users.order('number_of_wins desc').order('updated_at desc').take
+  def top_crawler
+    users.where('number_of_crawls > 0').order('number_of_crawls desc, updated_at desc').take
   end
 
   def worst_crawler
-    self.users.order('number_of_crawlings desc').order('updated_at desc').take
+    users.where('number_of_crawlings > 0').order('number_of_crawlings desc, updated_at desc').take
   end
 
   def longest_winning_streak
-    self.users.order('winning_streak desc').order('updated_at desc').take
+    users.where('winning_streak > 0').order('winning_streak desc, updated_at desc').take
+  end
+
+  def longest_winning_streak_ever
+    users.order('longest_winning_streak_games desc').take
   end
 
   def most_teams
-    self.users.sort_by(&:number_of_teams).last
+    users.sort_by(&:number_of_teams).last
   end
 
   def last_one
     active_user_ranking.last
   end
 
-  def longest_winning_streak_ever
-    self.users.order('longest_winning_streak_games desc').take
-  end
-
   def active_user_ranking
     users.reload.ranked.select(&:active?)
   end
 
   private
 
-    def sanitize_slug
-      self.slug = self.slug.downcase.parameterize
-    end
+  def sanitize_slug
+    self.slug = self.slug.downcase.parameterize
+  end
+
+  def ensure_user(user)
+    user || NullUser.new
```

Comment:
Nice. :) Just wanted to say, that `#top_crawler` etc. could be `nil`. Max already fixed that. :+1:
Instead of a whole new class, you could use a Struct here.